directory/directory_dest.go

```diff
@@ -54,49 +55,40 @@ func newImageDestination(sys *types.SystemContext, ref dirReference) (private.Im
 	// If directory exists check if it is empty
 	// if not empty, check whether the contents match that of a container image directory and overwrite the contents
 	// if the contents don't match throw an error
-	dirExists, err := pathExists(ref.resolvedPath)
-	if err != nil {
-		return nil, fmt.Errorf("checking for path %q: %w", ref.resolvedPath, err)
-	}
-	if dirExists {
-		isEmpty, err := isDirEmpty(ref.resolvedPath)
-		if err != nil {
+	dir, err := os.OpenFile(ref.resolvedPath, syscall.O_DIRECTORY|syscall.O_NOFOLLOW|os.O_RDONLY, 0)
```

Comment:
This code must remain cross-platform.

Reply:
Strictly speaking, neither syscall flags are needed here.
 - `O_DIRECTORY`  is not needed here since readdir below will fail anyway
 - `O_NOFOLLOW` is here to avoid security issues when a malicious symlink is placed in the directory. If we trust the directory, we can skip this.

Do you want me to drop these, or have these two flags used when on Unix only?

Reply:
Reading a bit more carefully: in the first place, it’s not defined at all that `syscall.O_` constants are valid in calls to `os.OpenFile`.

---

I don’t see what threat the `O_NOFOLLOW` is protecting against. [Opening a directory and using `*at` for all operations would be a thought, but that’s not what we are doing, and even then I don’t why `O_NOFOLLOW` would be necessary.]

So I’m fine with dropping both.

Reply:
Done that in https://github.com/containers/image/pull/2520